src/foam/demos/examples/services.fbe

```diff
@@ -0,0 +1,436 @@
+
+##  Advanced Service Patterns
+
+##   Service Decorators
+
+Services can be decorated for additional functionality:
+--
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'LoggingTaskServiceDecorator',
+  
+  implements: ['foam.demo.service.TaskService'],
+  
+  properties: [
+    {
+      class: 'Object',
+      name: 'delegate'
+    }
+  ],
+  
+  methods: [
+    {
+      name: 'createTask',
+      code: async function(title, description) {
+        log('LoggingDecorator: Creating task -', title);
+        var result = await this.delegate.createTask(title, description);
+        log('LoggingDecorator: Task created with ID -', result);
+        return result;
+      }
+    },
+    {
+      name: 'getTask',
+      code: async function(taskId) {
+        log('LoggingDecorator: Getting task -', taskId);
+        return await this.delegate.getTask(taskId);
+      }
+    },
+    {
+      name: 'deleteTask', 
+      code: async function(taskId) {
+        log('LoggingDecorator: Deleting task -', taskId);
+        return await this.delegate.deleteTask(taskId);
+      }
+    }
+  ]
+});
+
+// Demo decorator usage
+var baseService = foam.demo.service.SimpleTaskService.create();
+var decoratedService = foam.demo.service.LoggingTaskServiceDecorator.create({
+  delegate: baseService
+});
+
+var taskId = decoratedService.createTask('Decorated Task', 'With logging');
+
+##   Error Handling
+
+Services should handle errors gracefully:
+--
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'SafeTaskService',
+  extends: 'foam.demo.service.SimpleTaskService',
+  
+  methods: [
+    {
+      name: 'createTask',
+      code: function(title, description) {
+        try {
+          if (!title || title.trim().length === 0) {
+            throw new Error('Task title is required');
+          }
+          return this.SUPER(title, description);
+        } catch (error) {
+          log('Error creating task:', error.message);
+          throw error;
+        }
+      }
+    },
+    {
+      name: 'getTask',
+      code: function(taskId) {
+        try {
+          if (!taskId) {
+            throw new Error('Task ID is required');
+          }
+          return this.SUPER(taskId);
+        } catch (error) {
+          log('Error getting task:', error.message);
+          throw error;
+        }
+      }
+    }
+  ]
+});
+
+// Demo error handling
+var safeService = foam.demo.service.SafeTaskService.create();
+
+try {
+  safeService.createTask('', 'Empty title test');
+} catch (error) {
+  log('Caught expected error:', error.message);
+}
+
+
+
+##  Service Best Practices
+
+##   Lifecycle Management
+
+Services should properly manage resources:
+--
+foam.CLASS({
+  package: 'foam.demo.service', 
+  name: 'ManagedTaskService',
+  extends: 'foam.demo.service.SimpleTaskService',
+  
+  properties: [
+    {
+      name: 'isRunning',
+      class: 'Boolean',
+      value: false
+    }
+  ],
+  
+  methods: [
+    {
+      name: 'start',
```

Comment:
Again, start, stop may also not be good examples, as FOAM provides COREService interface which has start, stop, reload for server side models.  It could be extended to provide similar client side support - which, hithertoo, has not been required.